lib/services/role.go

```diff
@@ -2101,10 +2101,16 @@ func (set RoleSet) checkAccess(r AccessCheckable, mfa AccessMFAParams, matchers
 	additionalDeniedMessage := ""
 
 	var getRoleLabels func(types.Role, types.RoleConditionType) types.Labels
+	resourceHasLabels := true
+	getEmptyLabels := func(r types.Role, rct types.RoleConditionType) types.Labels { return types.Labels{} }
+
 	switch r.GetKind() {
 	case types.KindDatabase:
 		getRoleLabels = types.Role.GetDatabaseLabels
 		additionalDeniedMessage = "Confirm database user and name."
+	case types.KindDatabaseService:
```

Comment:
I'm trying to figure  out how why we need the RBAC model for `KindDatabaseService` resource if `DatabaseService` is the internal representation of db_service ? 
For instnace we have `KindDatabaseServer` resource  that is created  by healthcheck mechanism and we don't expose it like normal user resource.

Reply:
The requirements are that:
- Teleport's Database Service process heartbeats into the cluster
- End user can fetch all the Database Services

Given that, I tried to replicate what we had for `KindDatabaseServer`
One of functionality that I tried to mimic was the `ListResources` (to be used in the WebAPI).
To support `KindDatabaseService`, I had to change how we validate things here, specifically the labels.
There's no `databa_service_labels` or similar, so I changed how this part of the code works to accommodate for that scenario.

I guess Resources was only meant for Kinds that the User can view in the client (browser, for instance) 🤔
If that's the case, maybe I need to change some things

Reply:
Should we add labels to database service similar to WindowsDesktopService ?
https://github.com/gravitational/teleport/blob/e5d2b5a6c41c95f7bfe09791603ee6f8285dbbe5/lib/config/fileconf.go#L1909-L1912

Reply:
No strong opinion here
I think we might want to add labels to Services (db, kube, windows, ssh, ...) but for now it seems overkill

Reply:
@marcoandredinis 
In what conditions the access for  `DatabaseServices` should be denied. Right now I think that the `checkAccess` will always return true in case of ListResources. Could you cover this in UT ? 

Also don't we need to extend the proxy permission in 
 https://github.com/gravitational/teleport/blob/53e2d6cd0bb39a211343ff38bcd40f89948afb4c/lib/auth/permissions.go#L425  ?

Reply:
> In what conditions the access for DatabaseServices should be denied. Right now I think that the checkAccess will always return true in case of ListResources. Could you cover this in UT ?

`ListResources` always checks for `Read/List` access to the resource being listed, even before the call to `checkAccess`.
https://github.com/gravitational/teleport/blob/4387372e11b3c4f5aa858a7799ba733ada3f0e47/lib/auth/auth_with_roles.go#L1263

I added some tests here:
https://github.com/gravitational/teleport/pull/19469/files#diff-512af74d459e9975d43eb533b35f8a13a6a5bd911cf7d439010feab5e3aae914R2365


> Also don't we need to extend the proxy permission in

We do, but it's not necessary for the current PR
I'll add the extra permission on the next PR